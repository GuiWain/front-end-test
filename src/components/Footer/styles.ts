import styled, { css } from 'styled-components'
import { media } from 'styles/media'
import UseImage from 'next/image'

export const ImageISO = styled(UseImage).attrs({
  src: '/img/ISO_9001-2015.svg',
  alt: 'ISO 9001-2015',
  width: '35px',
  height: '35px'
})``

export const Footer = styled.footer`
  ${({ theme }) => css`
    background-color: ${theme.colors.secondaryMedium};
    width: 100%;
    left: 0;
    bottom: 0;
    padding: 1.5rem 0;
  `}
`

export const Description = styled.div`
  display: flex;
  margin: 0 auto;
  justify-content: center;
  align-items: center;
  gap: 1rem;
  margin: 0 auto 1.5rem;

  ${media.lessThan('tablet')`
    flex-direction: column;
  `}
`

export const Paragraph = styled.p`
  ${({ theme }) => css`
    text-align: center;
    font-weight: ${theme.font.light};
    font-size: ${theme.font.sizes.large};
    color: ${theme.colors.white};
    margin: 0;

    ${media.greaterThan('desktopFullHD')`
      font-size: ${theme.font.sizes.xlarge};
    `}

    ${media.lessThan('tablet')`
      width: 90%;
    `}
  `}
`

export const Span = styled.span`
  ${({ theme }) => css`
    width: 70%;
    display: block;
    margin: 0.5rem auto;

    text-align: center;
    font-weight: ${theme.font.light};
    font-size: ${theme.font.sizes.xxxsmall};
    color: ${theme.colors.white};

    ${media.greaterThan('desktopFullHD')`
      font-size: ${theme.font.sizes.xxsmall};
    `}

    ${media.lessThan('tablet')`
      margin: 1rem auto;
      width: 80%;
    `}
  `}
`
