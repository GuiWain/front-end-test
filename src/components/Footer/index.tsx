import * as S from './styles'

const Footer = () => (
  <S.Footer>
    <S.Description>
      <S.ImageISO />
      <S.Paragraph>
        Etiam dignissim diam quis enim lobortis scelerisque.
      </S.Paragraph>
    </S.Description>
    <S.Span>
      Id diam maecenas ultricies mi eget. Aenean sed adipiscing diam donec
      adipiscing.
    </S.Span>
    <S.Span>
      Phasellus faucibus scelerisque eleifend donec pretium vulputate sapien.
      Imperdiet dui accumsan sit amet nulla facilisi morbi tempus.
    </S.Span>
  </S.Footer>
)

export default Footer
