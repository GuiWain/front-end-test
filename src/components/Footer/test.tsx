import { render, screen } from 'utils/test-utils'

import Footer from '.'

describe('<Footer />', () => {
  it('should render the Footer', () => {
    const { container } = render(<Footer />)

    expect(screen.getByAltText(/ISO 9001-2015/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
