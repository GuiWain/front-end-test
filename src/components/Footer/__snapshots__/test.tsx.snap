// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the Footer 1`] = `
.c0 {
  background-color: #848080;
  width: 100%;
  left: 0;
  bottom: 0;
  padding: 1.5rem 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
  margin: 0 auto 1.5rem;
}

.c2 {
  text-align: center;
  font-weight: 300;
  font-size: 2.4rem;
  color: #FFFFFF;
  margin: 0;
}

.c3 {
  width: 70%;
  display: block;
  margin: 0.5rem auto;
  text-align: center;
  font-weight: 300;
  font-size: 1.2rem;
  color: #FFFFFF;
}

@media (max-width:768px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:1920px) {
  .c2 {
    font-size: 2.8rem;
  }
}

@media (max-width:768px) {
  .c2 {
    width: 90%;
  }
}

@media (min-width:1920px) {
  .c3 {
    font-size: 1.4rem;
  }
}

@media (max-width:768px) {
  .c3 {
    margin: 1rem auto;
    width: 80%;
  }
}

<footer
  class="c0"
>
  <div
    class="c1"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzUiIGhlaWdodD0iMzUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
          style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
        />
      </div>
      <img
        alt="ISO 9001-2015"
        class=""
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </div>
    <p
      class="c2"
    >
      Etiam dignissim diam quis enim lobortis scelerisque.
    </p>
  </div>
  <span
    class="c3"
  >
    Id diam maecenas ultricies mi eget. Aenean sed adipiscing diam donec adipiscing.
  </span>
  <span
    class="c3"
  >
    Phasellus faucibus scelerisque eleifend donec pretium vulputate sapien. Imperdiet dui accumsan sit amet nulla facilisi morbi tempus.
  </span>
</footer>
`;
